doc/api/dgram.md

```diff
@@ -64,8 +64,12 @@ function is passed a single `Error` object.
 added: v0.1.99
 -->
 
-The `'listening'` event is emitted whenever a socket begins listening for
-datagram messages. This occurs as soon as UDP sockets are created.
+The `'listening'` event is emitted on the first invocation of either `socket.send()` 
+or `socket.bind()` for the respective socket. 
+
+While the sockets are addressable immediately after creation using 
```

Comment:
What do you mean by "addressable"? https://github.com/nodejs/node/issues/23952#issuecomment-433901720 shows the socket not having an address until the send or bind.

Reply:
I meant that we have a socket object (returned by dgram.createSocket) on which we can attach listeners so its addressable in that sense. However Node.js creates the underlying OS socket only after `send` or `bind` are invoked. I did not mean addressable in the sense of the socket having an address.

Now I can see how using `addressable` introduces ambiguity. 

I was thinking to rephrase as follows

"While the different socket methods can be invoked immediately after creation using dgram.createSocket, Node.js doesn't create the underlying OS socket until ..."

But I'm a little confused because some socket methods can be used such as attaching listeners, `setTTL` etc, others such as `socket.address()` would throw an error until the underlying OS socket is created.

Let me know if you have any suggestions on the rephrasing and if my understanding is correct. 

Thanks :)

Reply:
I was thinking to remove the second para entirely (if the ambiguity cannot be removed) and leave just the first sentence which clarifies when the 'listening' event is emitted

"The `'listening'` event is emitted on the first invocation of either `socket.send()` 
or `socket.bind()` for the respective socket"

Reply:
@dsinecos In the interest of progress, how about we do that (remove the second paragraph), land this PR with just the one sentence changed, and then if you wish, you can open a second PR with this second paragraph added back in and it can be discussed/improved in that second PR?

Reply:
@Trott That sounds good. Allow me a couple of days. I'll make the suggested changes and update the PR. Thanks :)